In [1]:
train_path = "Train"

In [2]:
test_path = "Val"

In [5]:
import numpy as np
import matplotlib.pyplot as plt
import keras
from keras.layers import *
from keras.models import *
from keras.preprocessing import image

### CNN Based Model in Keras

In [15]:
model = Sequential()
model.add(Conv2D(32,(3,3),activation = 'relu',input_shape = (224,224,3)))
model.add(Conv2D(64,(3,3),activation='relu'))
model.add(MaxPooling2D(2,2))
model.add(Dropout(0.25))

model.add(Conv2D(64,(3,3),activation='relu'))
model.add(MaxPooling2D(2,2))
model.add(Dropout(0.25))


model.add(Conv2D(128,(3,3),activation='relu'))
model.add(MaxPooling2D(2,2))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(64,activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(1,activation='sigmoid'))
model.summary()


Model: "sequential_7"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
conv2d_8 (Conv2D)            (None, 222, 222, 32)      896       
_________________________________________________________________
conv2d_9 (Conv2D)            (None, 220, 220, 64)      18496     
_________________________________________________________________
max_pooling2d_5 (MaxPooling2 (None, 110, 110, 64)      0         
_________________________________________________________________
dropout_2 (Dropout)          (None, 110, 110, 64)      0         
_________________________________________________________________
conv2d_10 (Conv2D)           (None, 108, 108, 64)      36928     
_________________________________________________________________
max_pooling2d_6 (MaxPooling2 (None, 54, 54, 64)        0         
_________________________________________________________________
dropout_3 (Dropout)          (None, 54, 54, 64)       

In [18]:
model.compile(loss = 'binary_crossentropy',optimizer='adam',metrics = ['accuracy'])

## Train

In [28]:
train_datagen = image.ImageDataGenerator(
    rescale = 1./255,
    shear_range = 0.2,
    zoom_range = 0.2,
    horizontal_flip = True,
)



test_dataset = image.ImageDataGenerator(rescale=1./255)

In [29]:
train_generator = train_datagen.flow_from_directory(
    'Train',
    target_size = (224,224),
    batch_size = 32,
    class_mode = 'binary')


Found 224 images belonging to 2 classes.


In [21]:
train_generator.class_indices

{'Covid': 0, 'Normal': 1}

In [30]:
validation_generator = test_dataset.flow_from_directory(
    'Val',
    target_size = (224,224),
    batch_size = 32,
    class_mode = 'binary')

Found 60 images belonging to 2 classes.


In [32]:
hist = model.fit_generator(
    train_generator,
    steps_per_epoch=6,
    epochs = 10,
    validation_data = validation_generator,
    validation_steps=2
    
)

Epoch 1/10
6/6 [==============================] - 29s 5s/step - loss: 0.2177 - accuracy: 0.9271 - val_loss: 0.1145 - val_accuracy: 0.9667
Epoch 2/10
6/6 [==============================] - 28s 5s/step - loss: 0.2014 - accuracy: 0.9010 - val_loss: 0.1846 - val_accuracy: 0.9833
Epoch 3/10
6/6 [==============================] - 30s 5s/step - loss: 0.1516 - accuracy: 0.9531 - val_loss: 0.0843 - val_accuracy: 0.9667
Epoch 4/10
6/6 [==============================] - 32s 5s/step - loss: 0.1998 - accuracy: 0.9323 - val_loss: 0.1122 - val_accuracy: 0.9833
Epoch 5/10
6/6 [==============================] - 32s 5s/step - loss: 0.2218 - accuracy: 0.9219 - val_loss: 0.2180 - val_accuracy: 0.9667
Epoch 6/10
6/6 [==============================] - 31s 5s/step - loss: 0.1408 - accuracy: 0.9375 - val_loss: 0.0725 - val_accuracy: 0.9833
Epoch 7/10
6/6 [==============================] - 31s 5s/step - loss: 0.1589 - accuracy: 0.9271 - val_loss: 0.1254 - val_accuracy: 0.9833
Epoch 8/10
6/6 [==================

### Training Acc And Validation acc is Pretty Good
 -  98%
